overlord/hookstate/ctlcmd/is_connected.go

```diff
@@ -55,6 +68,17 @@ func init() {
 	})
 }
 
+func isConnectedPidCheckAllowed(info *snap.Info, plugOrSlot string) bool {
+	slot := info.Slots[plugOrSlot]
+	if slot != nil {
+		switch slot.Interface {
+		case "pulseaudio", "audio-record", "cups-control":
```

Comment:
all these interfaces may be deny-connection: true under some circumstances, which means sporting a slot doesn't mean the snap is intended/permitted to actually offer the interface.  Of course if the answer to "is connected" is true there's never problem disclosing that. As we discussed the issue is then whether we are worried about how much information we leak in the false cases,  the leakage is also different between the pid case and the apparmor label case, because in the latter case is very easy to establish if the other side is a snap at all or not without abusing this, the question then becomes about strict vs classic confinement.

Reply:
I am not sure what a malicious snap that provided an interface would be able to achieve with this in terms of information leakage - what kind of scenario are you envisaging here @pedronis? From what I can see here, this only allows a snap to query against it's own interfaces, so if a user has chosen to connect a snap to the malicious snap's interface, it should be assumed that the malicious snap can be aware of this. Can you elaborate?

Reply:
@alexmurray: with the pid based variant of this feature, it would let a snap enumerate the process ID space to check which ones belonged to snaps and which didn't.  That alone isn't a particularly useful, but is information that a snap wouldn't usually have and may enable some other attack we haven't thought of.

Reply:
The base apparmor profile already includes bits like:

```
  @{PROC}/@{pid}/ r,
...
  owner @{PROC}/@{pid}/cmdline r,
  owner @{PROC}/@{pid}/comm r,
```

This allows all snaps to enumerate processes running as their same user, and since it includes `cmdline` etc a snap can relatively easily identify other snap processes (running as the same user) by seeing if `/snap` is in `cmdline` - as such whilst we now allow snaps to identify other snaps regardless of the user, given most systems only have a very few number of users and most snaps will run as the primary user, I don't think this increases the information leakage in a significant manner. Nor am I aware of any way this enables any particular attack. 

One way to mitigate this would be to only permit this check if the specified `pid` has the same userid as the current `pid`, however, I suspect this would make this feature less useful to snaps like `pulseaudio` / `cups` etc which may run as root daemons etc.

So on balance, I am not concerned about this increased information leakage as a result.

Reply:
thanks for the input, I will mark this unblocked